pkg/kubelet/pleg/generic.go

```diff
@@ -224,6 +224,15 @@ func (g *GenericPLEG) relist() {
 		}
 	}
 
+	upcomingEventCount := 0
+	for _, value := range eventsByPodID {
+		upcomingEventCount = upcomingEventCount + len(value)
+	}
+	if (upcomingEventCount + len(g.eventChannel)) > cap(g.eventChannel) {
+		klog.Error("event channel is about to be full, discard this relist() cycle")
+		return
+	}
```

Comment:
Don't do this. Just:
```
select {
case g.eventChannel <- events[i]:
default:
  // log skipping event
}
```

Reply:
If we do this , we should update `g.podRecords`, and podRecord.old  will update to current pod. In next relist(), the events which get from compare `g.podRecords.current` and `g.podRecords.old` may be lost.